common/logger.go

```diff
@@ -104,3 +106,22 @@ func (l *Logger) Logf(level logrus.Level, category string, msg string, args ...i
 		l.lastLogCall = now
 	}
 }
+
+type mapAsObjectFormatter struct {
+	logrus.Formatter
+}
+
+func (f *mapAsObjectFormatter) Format(entry *logrus.Entry) ([]byte, error) {
+	if objects, ok := entry.Data["objects"].([]interface{}); ok {
+		var msg []string
+		for _, obj := range objects {
+			// TODO: Log error?
+			if o, err := json.Marshal(obj); err == nil {
+				msg = append(msg, string(o))
+			}
+		}
+		entry.Message = strings.Join(msg, " ")
+		delete(entry.Data, "objects")
+	}
+	return f.Formatter.Format(entry)
+}
```

Comment:
By default it would marshall it to an array of objects, which would look strange when logged. We could write our own `MarshalJSON()` implementation, but I think that would work similarly to this and be more code.

I prefer the original version to your suggestion, sorry. It has one more level of nesting, but it's shorter and calls `f.Formatter.Format()` once.

Reply:
No problem :)